src/main/resources/interface-specification.yml

```diff
@@ -737,85 +700,50 @@ components:
       schema:
         type: string
   schemas:
-    KeysResponse:
-      type: object
-      properties:
-        keys:
-          type: array
-          items:
-            $ref: '#/components/schemas/ClientKey'
-      required:
-        - keys
-    Keys:
-      type: array
-      items:
-        $ref: '#/components/schemas/KeySeed'
-    KeySeed:
-      description: 'Models the seed for a public key to be persisted'
+    Key:
+      description: 'Models the PersistentKey'
       type: object
       properties:
         relationshipId:
           type: string
           format: uuid
           description: 'Represents the identifier of the party relationship whose platform security operator is uploading the key'
-        key:
-          type: string
-          description: 'Base64 UTF-8 encoding of a public key in PEM format'
-        createdAt:
-          type: string
-          format: date-time
-        use:
-          $ref: '#/components/schemas/KeyUse'
-        alg:
+        kid:
           type: string
-          description: 'The algorithm type of the key.'
         name:
           type: string
-          description: 'Descriptive name of this key.'
-      required:
-        - relationshipId
-        - key
-        - use
-        - alg
-        - name
-        - createdAt
-    ClientKey:
-      type: object
-      properties:
-        key:
-          $ref: '#/components/schemas/Key'
-        relationshipId:
+        encodedPem:
           type: string
-          format: uuid
-        name:
+        algorithm:
           type: string
+        use:
+          $ref: '#/components/schemas/KeyUse'
         createdAt:
           type: string
           format: date-time
       required:
-        - key
         - relationshipId
+        - kid
         - name
+        - encodedPem
+        - algorithm
+        - use
         - createdAt
+    Keys:
+      type: array
```

Comment:
same, we can use an object containing the array.
Probably the task description was misleading, my bad